### ISSUE TITLE
Show character positions on semantic errors

### ISSUE THREAD

**Issue by malte-v** [google/cel-cpp#48]
When the expression is syntactically incorrect, we get a helpful error message:
```
ERROR: <input>:1:33: Syntax error: token recognition error at: '"'
 | unmatched_quotation_mark_follows"
 | ................................^
```
When there is a semantic error, we get a message but no character position:

`"foo" + 5`

`No matching overloads found`

Is there some way to get this information from semantic errors? Thanks!

**Comment by kyessenov** [comment]
Yes, this is a good idea to add positions to errors. Historically, the evaluator was operating on expressions with stripped source info. Now that we have a parser, we should make use of the supplied source info.
